core/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java

```diff
@@ -180,7 +180,7 @@ protected XContentBuilder newXContentBuilder(XContentType type, OutputStream str
      * the state.
      */
     public final T read(Path file) throws IOException {
-        try (Directory dir = newDirectory(file.getParent())) {
+        try (Directory dir = newReadOnlyDirectory(file.getParent())) {
```

Comment:
that won't eliminate the race completely. If we check that the directory exists and then open that directory but the delete has happened in-between these two actions we resurrect the directory again :-(